aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -1011,6 +1022,75 @@ module aptos_token::token {
         );
     }
 
+
+    /// Burn a token by creator when the token's BURN_BY_CREATOR is true
+    /// The token is owned at address owner
+    public entry fun burn_by_creator(
+        creator: &signer,
+        owner: address,
+        collection: String,
+        name: String,
+        property_version: u64,
+        amount: u64,
+    ) acquires Collections, TokenStore {
+        let creator_address = signer::address_of(creator);
+        assert!(amount > 0, error::invalid_argument(ENO_BURN_TOKEN_WITH_ZERO_AMOUNT));
+        let token_id = create_token_id_raw(creator_address, collection, name, property_version);
+        let creator_addr = token_id.token_data_id.creator;
+        assert!(
+            exists<Collections>(creator_addr),
+            error::not_found(ECOLLECTIONS_NOT_PUBLISHED),
+        );
+
+        let collections = borrow_global_mut<Collections>(creator_address);
+        assert!(
+            table::contains(&collections.token_data, token_id.token_data_id),
+            error::not_found(ETOKEN_DATA_NOT_PUBLISHED),
+        );
+
+        let token_data = table::borrow_mut(
+            &mut collections.token_data,
+            token_id.token_data_id,
+        );
+
+        // for creator burn token, the property should be explicitly set in the property_map.
+        assert!(
+            property_map::contains_key(&token_data.default_properties,  &string::utf8(BURN_BY_CREATOR)),
+            error::permission_denied(ETOKEN_CANNOT_BURN_BY_CREATOR)
+        );
+
+        let burn_by_owner_flag = property_map::read_bool(&token_data.default_properties, &string::utf8(BURN_BY_CREATOR));
```

Comment:
If the token's properties are mutable, the creator can change either BURN_BY_OWNER or BURN_BY_CREATOR anytime right?

Reply:
only the properties stored at token could be mutable. Properties stored at TokenData are immutable so that all the token's properties will start from the same set of values. This is written in the standard.